### PR TITLE
Add information on downloading historical results

### PR DESCRIPTION
**Pull request by dstrain115**
- Adds documentation to engine docs on how to get previous
results using the program and job id.

### CODE REVIEW

File: docs/google/engine.md [hunk 1]
```diff
@@ -191,3 +191,51 @@ for b in range(num_circuits_in_batch):
...
+In order to retrieve previous results, you will need the program id and the
+job id.  This can be retrieved from the job object when you run a sweep.
+Currently, getting the program and job ids can only be done through the
+`Engine` interface and not through the sampler interface.
+You can then use `get_program` and `get_job` to retrieve the results.
+See below for an example:
+
+```
+# Initialize the engine object
+engine = cirq.google.Engine(project_id='YOUR_PROJECT_ID')
```
Comment (mrwojtek) [inline docs/google/engine.md:207]: I would also mention that results are restricted to the project; only project collaborators are allowed to access the data (so people are not mislead this is a public database).

File: docs/google/engine.md [hunk 2]
```diff
@@ -191,3 +191,51 @@ for b in range(num_circuits_in_batch):
...
+#Create an example circuit
+qubit = cirq.GridQubit(5, 2)
+circuit = cirq.Circuit(
+    cirq.X(qubit)**sympy.Symbol('t'),
+    cirq.measure(qubit, key='result')
+)
+param_sweep = cirq.Linspace('t', start=0, stop=1, length=10)
+
+# Run the circuit
+job = e.run_sweep(program=circuit, params=param_sweep1, repetitions=1000, processor_ids=[PROCESSOR_ID], gate_set=GATE_SET)
```
Comment (mrwojtek) [inline docs/google/engine.md:218]: param_sweep1 -> param_sweep

80 character line limit?

File: docs/google/engine.md [hunk 3]
```diff
@@ -191,3 +191,51 @@ for b in range(num_circuits_in_batch):
...
+Currently, getting the program and job ids can only be done through the
+`Engine` interface and not through the sampler interface.
+You can then use `get_program` and `get_job` to retrieve the results.
+See below for an example:
+
+```
+# Initialize the engine object
+engine = cirq.google.Engine(project_id='YOUR_PROJECT_ID')
+
+#Create an example circuit
```
Comment (mrwojtek) [inline docs/google/engine.md:209]: "# Create"

Comment (dstrain115) [reply docs/google/engine.md:209]: Done.

File: docs/google/engine.md (hunk 2 above)
Comment (dstrain115) [reply docs/google/engine.md:218]: Changed.

File: docs/google/engine.md (hunk 1 above)
Comment (dstrain115) [reply docs/google/engine.md:207]: Added this in the beginning of the section.

**Review by mrwojtek — APPROVED** [review]

File: docs/google/engine.md [hunk 4]
```diff
@@ -191,3 +191,57 @@ for b in range(num_circuits_in_batch):
      idx+=1
 ```
 
+
+## Downloading historical results
+
+Results from  previous computations are archived and can be downloaded later
+by those in the same cloud project.  You must be use the same project id to
```
Comment (balopat) [inline docs/google/engine.md:198]:
```suggestion
by those in the same cloud project.  You must use the same project id to
```

**Review by balopat — APPROVED** [review]
LGTM with one typo

File: docs/google/engine.md [hunk 5]
```diff
@@ -191,3 +191,56 @@ for b in range(num_circuits_in_batch):
      idx+=1
 ```
 
+
+## Downloading historical results
+
+Results from  previous computations are archived and can be downloaded later
+by those in the same cloud project.  You must use the same project id to
+access historical results or your request will be denied.
```
Comment (wcourtney) [inline docs/google/engine.md:199]: What do you think of adding a little motivation for the "same project" requirement by explaining that program IDs are unique within a project?

File: docs/google/engine.md [hunk 6]
```diff
@@ -191,3 +191,56 @@ for b in range(num_circuits_in_batch):
...
+In order to retrieve previous results, you will need the program id and the
+job id.  This can be retrieved from the job object when you run a sweep.
+Currently, getting the program and job ids can only be done through the
```
Comment (wcourtney) [inline docs/google/engine.md:202]: I realize that this is a documentation PR, but still think it's worth calling that that we really should have "list" methods as well, but they don't exist in Cirq today.

File: docs/google/engine.md [hunk 7]
```diff
@@ -191,3 +191,56 @@ for b in range(num_circuits_in_batch):
...
+
+# Recreate the job object
+historical_job = engine.get_program(program_id=program_id).get_job(job_id=job_id)
+
+# Retrieve the results
+historical_results = historical_job.results()
+
+```
+
+If you did not save the ids, you can still find them from your
```
Comment (wcourtney) [inline docs/google/engine.md:245]: Is it also worth pointing out that you can provide your own program and job IDs? If you use something meaningful to your script (e.g. an index in an iterative process), then you could generate the ids yourself if you need to look them up later.

File: docs/google/engine.md (hunk 6 above)
Comment (dstrain115) [reply docs/google/engine.md:202]: Sure.  Want to create a bug for it and I'll go ahead and assign it to me?

File: docs/google/engine.md (hunk 5 above)
Comment (dstrain115) [reply docs/google/engine.md:199]: Added a bit about this later.  See what you think.

File: docs/google/engine.md (hunk 7 above)
Comment (dstrain115) [reply docs/google/engine.md:245]: Clarified this.  See what you think.

File: docs/google/engine.md (hunk 6 above)
Comment (wcourtney) [reply docs/google/engine.md:208]: Added https://github.com/quantumlib/Cirq/issues/3302

File: docs/google/engine.md (hunk 5 above)
Comment (wcourtney) [reply docs/google/engine.md:199]: LGTM, thanks!

**Review by wcourtney — APPROVED** [review]

**Pull request merged**